src/mym.cpp

```diff
@@ -816,13 +817,41 @@ void mexFunction(int nlhs, mxArray*plhs[], int nrhs, const mxArray*prhs[]) {
             if (pcmp!=NULL)
                 mxFree(pcmp);
         }
-        else {
-            // check that no placeholders are present in the query
-            char*p_tmp_o = strstr(query, PH_OPEN);
-            char*p_tmp_c = strstr(query, PH_CLOSE);
-            if (p_tmp_o||p_tmp_c)
-                mexErrMsgTxt("The query contains placeholders, but no additional arguments!");
+        // Check that no placeholders are present in the query except for \{ and \}
+        char* parsedQueryString = (char*)mxCalloc(strlen(query), sizeof(char)); // Query string with escape character removed
+        size_t parsedQueryStringCurrentIndex = 0;
+
+        // Loop through the query string character by character
+        for (size_t i = 0; query[i] != '\0'; i++) {
+            
+            if ((query[i] == '{' || query[i] == '}') && i != 0) {

```

Comment:
Perhaps we can use std libraries for `regex` to simplify a bit the operations that are performed. So we don't have to introduce custom parsing and character indexing here.